### PR TITLE
feat(vm): add force stop

### PR DESCRIPTION
**Pull request by FrankYang0529**
**Feature:**
If users set `terminationGracePeriodSeconds`, we need provide a way to force stop the VM.

**Related Issue:**
https://github.com/harvester/harvester/issues/2466

**Test plan:**

Case 1: If users don't set `default-termination-grace-period-seconds` setting, `terminationGracePeriodSeconds` in a VM is 120 seconds

1. Create a VM.
2. Check whether `spec.template.spec.terminationGracePeriodSeconds` in the VM is `120`.

Case 2: `default-termination-grace-period-seconds` setting can't be negative.

1. Set value in `default-termination-grace-period-seconds` setting as `"-1"`. The webhook should return an error.

Case 3: `default-termination-grace-period-seconds` setting should be able to be parsed as an integer.

1. Set value in `default-termination-grace-period-seconds` setting as `"not an integer"`. The webhook should return an error.

Case 4: Set `default-termination-grace-period-seconds` setting and `terminationGracePeriodSeconds`in a new VM should follow the setting.

1. Set value in `default-termination-grace-period-seconds` setting as `"60"`.
2. Create a VM.
3. Check whether `spec.template.spec.terminationGracePeriodSeconds` in the VM is `60`.

Case 5: The controller should wait for `terminationGracePeriodSeconds`, before really removing the VM.

1. Follow Case 4.
2. SSH to the VM and run following commands.
```
# sudo vim /etc/systemd/system/set_gracefulshutdown.service
[Unit]
Description=Set flag for graceful shutdown
DefaultDependencies=no
RefuseManualStart=true
Before=shutdown.target

[Service]
Type=oneshot
TimeoutStartSec=50
ExecStart= /bin/bash -c 'sleep infinity'

[Install]
WantedBy=shutdown.target

# sudo systemctl daemon-reload
# sudo systemctl enable set_gracefulshutdown
```
3. On VM dashboard, stop the VM. We can see OS will wait for 170 seconds and eventually shutdown at 180 seconds:
Screenshot 2023-05-18 at 3 45 46 PM

Case 6: Users can force stop the VM if they don't wait for a graceful shutdown.

1. Follow Case 5.
2. Enable Developer Tools & Features.
3. Start the VM again.
4. Stop the VM again.
5. Click "View in API" on the VM.
6. Click "forceStop" and the controller will stop the VM immediately.

### CODE REVIEW

File: pkg/api/vm/formatter.go [hunk 1]
```diff
@@ -284,3 +289,14 @@ func (vf *vmformatter) getVMI(vm *kubevirtv1.VirtualMachine) *kubevirtv1.Virtual
 	}
 	return nil
 }
+
+func (vf *vmformatter) canForceStop(vm *kubevirtv1.VirtualMachine) bool {
+	if vm == nil {
+		return false
+	}
+
+	if vm.Status.PrintableStatus != kubevirtv1.VirtualMachineStatusStopping {
```
Comment (bk201) [inline pkg/api/vm/formatter.go:314]: Do we also have the need to force stop a running VM?

Comment (guangbochen) [reply pkg/api/vm/formatter.go:314]: ~~should we check if the VM is not stopped? also force stop a VM with stopping status should be supported.~~

File: pkg/webhook/resources/virtualmachine/mutator.go [hunk 2]
```diff
@@ -373,3 +378,27 @@ func (m *vmMutator) addNodeSelectorTerms(defaultNamespace string, nodeSelector *
 
 	return nil
 }
+
+func (m *vmMutator) patchTerminationGracePeriodSeconds(vm *kubevirtv1.VirtualMachine, patchOps types.PatchOps) (types.PatchOps, error) {
```
Comment (guangbochen) [inline pkg/webhook/resources/virtualmachine/mutator.go:383]: seems there is no global config on the kubevirt CR?

Comment (FrankYang0529) [reply pkg/webhook/resources/virtualmachine/mutator.go:383]: Yeah, I don't find global config on KubeVirt CR.

**Review by futuretea — APPROVED** [review]
LGTM, Thanks!

**Review by bk201 — APPROVED** [review]
LGTM!

**Review by guangbochen — APPROVED** [review]

**Pull request merged**